runtime/commonTest/src/kotlinx/serialization/json/polymorphic/JsonDeserializePolymorphicTwiceTest.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json.polymorphic
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.*
+import kotlinx.serialization.test.*
+import kotlin.test.*
+
+class JsonDeserializePolymorphicTwiceTest {
+
+    @Serializable
+    sealed class Foo {
+        @Serializable
+        data class Bar(val a: Int) : Foo()
+    }
+
+    @Test
+    fun testDeserializeTwice() = noNative { // #812
+        val json = Json.toJson(Foo.Bar(1) as Foo)
```

Comment:
I'd rather use `Json.toJson(Foo.serializer(), Foo.Bar(1))` instead of ignoring test on Native completely